integration/addr/manager.go

```diff
@@ -1,12 +1,39 @@
 package addr
 
 import (
+	"fmt"
 	"net"
+	"sync"
+	"time"
 )
 
-// Suggest suggests a address a process can listen on. It returns
-// a tuple consisting of a free port and the hostname resolved to its IP.
-func Suggest() (port int, resolvedHost string, err error) {
+type portCache struct {
+	lock  sync.Mutex
+	ports map[int]time.Time
+}
+
+func (c *portCache) add(port int) bool {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	// remove outdated port
+	for p, t := range c.ports {
+		if time.Since(t) > 1*time.Minute {
```

Comment:
nit: this is some random number/duration. Can we maybe pull that up into a `const`?

Reply:
Done